lib/index.js

```diff
@@ -167,14 +167,21 @@ internals.root = function () {
         }
     };
 
-    root.assert = function (value, schema, message) {
+    root.assert = function (...args) {
 
-        this.attempt(value, schema, message);
+        this.attempt(...args);
     };
 
-    root.attempt = function (value, schema, message) {
+    root.attempt = function (value, schema, ...args/* [message], [options]*/) {
```

Comment:
One thought: how about making it an `[message | options]` like in [string regex](https://github.com/hapijs/joi/blob/v14.3.1/API.md#stringregexpattern-name--options) function? This would allow to use `validate` options AND the custom message. The third argument would be then either `'custom message'` or `{ message: 'custom message', ...otherValidateOptions }`.

What do you think about that?

Reply:
I implemented it in a way that `message`, `options`, both, or neither are acceptable. I even added a couple of tests around using both `message` and `options` in a few different scenarios.

As far as following the `regex()` param usage pattern, that's definitely an option. I was trying to keep the impact on current usage as minimal as possible, because `attempt` is actually used in quite a few places by internals behind the scenes.

Reply:
Yeah, I think we can keep this implementation. I just wanted to share my thought :)